packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,228 @@
+import { Pagination, PaginationVariant } from '@patternfly/react-core/dist/js/components/Pagination';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/PrimaryToolbar';
+import flatten from 'lodash/flatten';
+import debounce from 'lodash/debounce';
+import { calculateMeta, calculateActiveFilters, createRows } from './helpers';
+
+class RuleTable extends Component {
+    state = {
+        filterValues: {},
+        expanded: []
+    }
+
+    debounceFetchData = debounce(options => {
+        const { fetchData } = this.props;
+        fetchData(options);
+    }, 500)
+
+    onUpdate = (isDebounce = false) => {
+        const { fetchData, rules, columns, sortBy, detail } = this.props;
+        const { filterValues } = this.state;
+        const method = isDebounce ? this.debounceFetchData : fetchData;
+        method({
+            meta: calculateMeta(rules.meta),
+            filterValues,
+            sortBy: { ...sortBy, column: columns[sortBy.index - Boolean(detail)] }
+        });
+    }
+
+    onFilterDelete = (_e, [ deleteItem ], isAll) => {
+        const { filterValues } = this.state;
+        if (isAll) {
+            this.setState({ filterValues: {} }, () => this.onUpdate());
+        } else {
+            if (Array.isArray(filterValues[deleteItem.type])) {
+                const activefilter = filterValues[deleteItem.type];
+                const [ chipToDelete ] = deleteItem.chips;
+                this.setState({ filterValues: {
+                    ...filterValues,
+                    [deleteItem.type]: activefilter.filter((item) => item !== chipToDelete.value)
+                } }, () => this.onUpdate());
+            } else {
+                delete filterValues[deleteItem.type];
+                this.setState({ filterValues }, () => this.onUpdate());
+            }
+        }
+    }
+
+    onCollapse = (_e, key, isOpen, { ruleId }) => {
+        const { expanded } = this.state;
+        const collapseKey = ruleId || key;
+        if (isOpen) {
+            this.setState({
+                expanded: [
+                    ...expanded,
+                    collapseKey
+                ]
+            });
+        } else {
+            const currIndex = expanded.findIndex(id => id === collapseKey);
+            expanded.splice(currIndex, 1);
+            this.setState({ expanded });
+        }
+    }
+
+    onFilterChange = (type, _e, value) => {
+        this.setState({
+            filterValues: {
+                ...this.state.filterValues,
+                [type]: value
+            }
+        }, () => this.onUpdate(true));
+    }
+
+    onSort = (_e, index, direction) => {
+        const { fetchData, rules, columns, detail } = this.props;
+        const { filterValues } = this.state;
+        fetchData({
+            meta: calculateMeta(rules.meta),
+            filterValues,
+            sortBy: { index, direction, column: columns[index - Boolean(detail)] }
```

Comment:
Why to use here `index - Boolean(detail)`? Maybe better `index - Number(!!detail)`?

Reply:
What do you think about using `const offset = detail ? -1 : 0;`?